### PR TITLE
Brian Greeson

### PR DESCRIPTION
**Pull request by brian-greeson**
## Done with user story 3
 - I saved the prices as strings out of curiosity.
 - I originally wanted to have my average_prices method work on`machine.snacks.average_price` but I couldn't figure out where to put the method since `.snacks` is an AR relation class.  I started thinking of putting it in the snacks model as an instance method and making a call to `super` to get the machine that was calling the snack.  That didn't seem quite right so i just put it in the machine model and called it done.  I would be curious how to implement this.

### CODE REVIEW

File: app/views/snack/show.html.erb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<h1><%= @snack.name %></h1>
+<h2><%= @snack.price %></h2>
+<section class="snack-locations">
+  <h3>Available at:</h3>
+  <% @snack.machines.each do |machine| %>
+    <p><%= machine.location %> (<%= machine.snacks.count %> kinds of snacks, average price of $<%= machine.average_snack_price %>) /p>
```
Comment (brian-greeson) [inline app/views/snack/show.html.erb:6]: Oups!! i totally put an AR call in the view here `.count` :(

Comment (megstang) [reply app/views/snack/show.html.erb:6]: 😏 Good catch though

File: app/models/machine.rb [hunk 2]
```diff
@@ -2,4 +2,10 @@ class Machine < ApplicationRecord
   validates_presence_of :location
   
   belongs_to :owner
+  has_many :machine_snacks
+  has_many :snacks, through: :machine_snacks
+
+  def average_snack_price
+    snacks.average('price::float').to_s
+  end
```
Comment (megstang) [inline app/models/machine.rb:10]: did you try something like what i wrote below for a class method on the snack model (referring to your comment)?
```ruby
def self.average_price
   average('price::float').to_s
end 
```

File: app/views/machines/show.html.erb [hunk 3]
```diff
@@ -1 +1,9 @@
 <h1><%= @machine.location %> Vending Machine</h1>
+<section class="snack-list">
+<% @machine.snacks.each do |snack| %>
+  <p><%= snack.name %>: $<%=  snack.price %></p>
+<% end %>
+</section>
+<section class="snack-stats">
+  <p>Average snack price: $<%= @machine.average_snack_price.to_f.round(2) %></p>
```
Comment (megstang) [inline app/views/machines/show.html.erb:8]: you can look into the action view helper, `number_to_currenct` for formatting this output.

File: app/views/snack/show.html.erb (hunk 1 above)
Comment (megstang) [reply app/views/snack/show.html.erb:6]: 😏 Good catch though

**Review by megstang — COMMENTED** [review]
@brian-greeson Good job on this! I wasn't totally sure what you were asking about in your comment. I tried addressing it, but if I didn't fully answer what you're looking for, let me know!

**Pull request closed without merging**